File: python/test/models/keyboard_event_test.py
import unittest
from ...models.keyboard_event import *
from ...models.event import *


class KeyboardTest(unittest.TestCase):
    def test_keyboard_event(self):
        event = Event.parse(
            {
                "type": "keyboard",
                "key": "a",
                "state": "press",
                "specialKey": "enter",
            }
        )
        self.assertEqual(type(event), KeyboardEvent, "Event type is not KeyboardEvent")
        self.assertEqual(event.key, "a")
        self.assertEqual(event.state, KeyboardButtonState.press)
        self.assertEqual(event.specialKey, SpecialKey.enter)

        eventJson = event.toJson()
        self.assertEqual(eventJson["type"], KeyboardEvent.eventType)
        self.assertEqual(eventJson["key"], "a")
        self.assertEqual(eventJson["state"], KeyboardButtonState.press)
        self.assertEqual(eventJson["specialKey"], SpecialKey.enter)


if __name__ == "__main__":
    unittest.main()

File: test/models/keyboard_event_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:steps_recorder/models/events/event.dart';
import 'package:steps_recorder/models/events/keyboard/keyboard_event.dart';
import 'package:steps_recorder/models/events/keyboard/special_keys.dart';

void main() {
  group('Keyboard Event', () {
    test('fromJson & toJson', () {
      Event event = Event.parse({
        'type': 'keyboard',
        'key': 'a',
        'state': 'press',
        'specialKey': 'enter',
      });
      if (event is! KeyboardEvent) {
        fail('Event is not a KeyboardEvent');
      }
      expect(event.key, 'a');
      expect(event.state, KeyboardButtonState.press);
      expect(event.specialKey, SpecialKey.enter);

      final json = event.toJson();
      expect(json['key'], 'a');
      expect(json['state'], 'press');
      expect(json['specialKey'], 'enter');
    });
  });
}
